app/controllers/concerns/api_actionable.rb

```diff
@@ -0,0 +1,84 @@
+module ApiActionable
+  extend ActiveSupport::Concern
+  included do
+    before_action :initialize_api_actions, only: [:update, :show, :destroy]
+    before_action :check_for_redirect_action, only: [:create, :update, :show, :destroy]
+    after_action :execute_api_actions, only: [:show, :create, :update, :destroy]
+    before_action :check_for_serve_file_action, only: [:show, :create, :update, :destroy]
+    rescue_from StandardError, with: :handle_error
+  end
+
+
+  def check_for_redirect_action
+    @redirect_action = @api_namespace.send(api_action_name).where(action_type: 'redirect').last
+  end
+
+  def check_for_serve_file_action
+    serve_file_action = @api_namespace.send(api_action_name).where(action_type: 'serve_file').last
+    return if serve_file_action.nil?
+
+    serve_file_action.update(lifecycle_stage: 'executing')
+    file_id = helpers.file_id_from_snippet(serve_file_action.file_snippet)
+    file = Comfy::Cms::File.find(file_id)
+    if params[:action] == 'show' && @redirect_action.nil?
+      flash.now[:file_url] = rails_blob_url(file.attachment)
+    else
+      flash[:file_url] = rails_blob_url(file.attachment)
+    end
+    serve_file_action.update(lifecycle_stage: 'complete', lifecycle_message: file.label)
```

Comment:
passing a string like `label: #{file.label}  id: #{file.id} mime_type: #{file.mime_type}`  would be better here. Don't quote me on the mime type part of the snippet :laughing: